blocks/rich-text/index.js

```diff
@@ -366,21 +369,12 @@ export class RichText extends Component {
 		}
 	}
 
-	fireChange() {
-		this.savedContent = this.getContent();
-		this.editor.save();
-		this.props.onChange( this.state.empty ? [] : this.savedContent );
-	}
-
 	/**
 	 * Handles any case where the content of the tinyMCE instance has changed.
 	 */
 	onChange() {
-		// Note that due to efficiency, speed and low cost requirements isDirty may
-		// not reflect reality for a brief period immediately after a change.
-		if ( this.editor.isDirty() ) {
-			this.fireChange();
-		}
+		this.savedContent = this.state.empty ? [] : this.getContent();
```

Comment:
@aduth - should we extract `[]` to its own shared variable to avoid having always one reference to compare? I wanted to double check - you know better all those perf bits. It might not make any sense in this context :)